tsinfer/formats.py

```diff
@@ -1116,7 +1116,7 @@ def add_site(
         :return: The ID of the newly added site.
         :rtype: int
         """
-        genotypes = np.array(genotypes, dtype=np.uint8, copy=False)
```

Comment:
I don't think we should do this --- this'll be a significant extra cost here when we're building samples files. I don't mind if we're not quite as general in the inputs, this is a performance critical step.

Reply:
To be clear, let's keep it as ``genotypes = np.array(genotypes, dtype=np.int8, copy=False)``

Reply:
Ah, ISWYM. But I think the safe_np_int_cast will (if given an array of `np.uint8`) not make a copy, but simply return the array as-is. It calls

```
return input_array.astype(dtype, casting='safe', copy=copy)
```

and I'm assuming that if dtype is already the same as input_array.dtype, it will return the original array, no?

Reply:
In other words, if the user wants to be fast, they can submit the data as an array of dtype `np.int8` and it should be just as fast as it is at present. I think...

Reply:
I see, the checks only occur when the same call we're making anyway fails. OK, you're right, let's use ``safe_np_int_cast``. We still need ``copy=False`` though.

Reply:
Agree, we need copy=False, but this is the default for `safe_np_int_cast()`

Reply:
Right, doh!

Reply:
(NB I set the default to copy = False due to your comment at https://github.com/tskit-dev/tskit/pull/245#discussion_r296316747)

Reply:
Yes, sorry, I forgot about this.